File: ecs_vehicles/doctype_triggers/stock/stock_entry/stock_entry.py
from __future__ import unicode_literals
import frappe
from frappe import _


@frappe.whitelist()
def before_insert(doc, method=None):
    pass

@frappe.whitelist()
def get_item_table(doc, method=None):
    # frappe.throw(str(doc))
    # doc.items = []
    item = frappe.db.sql(""" select a.item_group,
                                    a.maintenance_type,
                                    b.name,
                                    b.vehicles,
                                    b.vehicle_no,
                                    b.vehicle_brand,
                                    b.entity_name,
                                    b.vehicle_model,
                                    a.item_code,
                                    a.item_name,
                                    a.default_unit_of_measure,
                                    a.brand,
                                    a.description,
                                    a.qty
                            from `tabMaintenance Order Item` a join `tabMaintenance Order` b
                            on a.parent = b.name
                            where b.name = '{maintenance_order}'
                        """.format(maintenance_order=doc), as_dict=1)
    return item
   
    # for x in item:
    #     table = doc.append("items", {})
    #     table.item_group = x.item_group
    #     table.item_code = x.item_code
    #     table.item_name = x.item_name
    #     table.uom = x.default_unit_of_measure
    #     table.qty = x.qty
    #     table.description = x.description

@frappe.whitelist()
def after_insert(doc, method=None):
    pass
@frappe.whitelist()
def onload(doc, method=None):
    pass
@frappe.whitelist()
def before_validate(doc, method=None):
    pass
@frappe.whitelist()
def validate(doc, method=None):
    pass
@frappe.whitelist()
def on_submit(doc, method=None):
    pass
@frappe.whitelist()
def on_cancel(doc, method=None):
    pass
@frappe.whitelist()
def on_update_after_submit(doc, method=None):
    pass
@frappe.whitelist()
def before_save(doc, method=None):
    pass
@frappe.whitelist()
def before_cancel(doc, method=None):
    pass
@frappe.whitelist()
def on_update(doc, method=None):
    pass

File: ecs_vehicles/doctype_triggers/stock/stock_entry/stock_entry.js

// frappe.ui.form.on('Stock Entry', {
//  maintenance_order: function(frm) {
// 			frappe.call({
// 				doc: frm.doc,
//                 method: "ecs_vehicles.ecs_vehicles.doctype_triggers.stock.stock_entry.stock_entry.get_item_table",
// 				callback: function(r) {
// 					frm.refresh_fields();
// 					frm.refresh();
// 				}
// 			});
// 	}
// });
// maintenance_order: function(frm){   // I have used refresh you can use any trigger
//     frm.clear_table('items');
//     frappe.call({
//         method:"ecs_vehicles.doctype_triggers.stock.stock_entry.stock_entry.get_item_table",
//         freeze: true,
//         args:{
//             doc:frm.doc.maintenance_order,
//            },
//             callback: function (response) {
//             if (response.message) {
//                 $.each(response.message, function(i,row) {   // row can be anything, it is merely a name


//                         var child_add = cur_frm.add_child("items");  // child_add can be anything
//                         child_add.item_group = row.item_group;
//                         child_add.item_code = row.item_code; // you can add as many fields as you want
//                         child_add.item_name = row.item_name; // you can add as many fields as you want
//                         child_add.uom = row.default_unit_of_measure; // you can add as many fields as you want
//                         child_add.qty = row.qty; // you can add as many fields as you want
//                         child_add.description = row.description; // you can add as many fields as you want
                      

//                             });
//           frm.refresh_fields("items");
//             }
//             }
//             });
//         }


